libs/core/src/lib/combobox/combobox.component.html

```diff
@@ -6,38 +6,58 @@
+                   (keyup)="onInputKeyupHandler($event)"
+                   [disabled]="disabled"
+                   [(ngModel)]="inputText"
+                   (ngModelChange)="handleSearchTermChange()"
+                   [placeholder]="placeholder">
+        </fd-input-group>
     </fd-popover-control>
     <fd-popover-body *ngIf="displayedValues && displayedValues.length">
-        <fd-menu class="fd-combobox-input-menu-overflow" [style.maxHeight]="maxHeight">
+        <ul fd-list [dropdownMode]="true" class="fd-combobox-input-menu-overflow" [style.maxHeight]="maxHeight" [hasMessage]="listMessages.length > 0">
             <ng-content></ng-content>
-            <ul fd-menu-list>
+            <ng-container *ngIf="groupFn">
+                <ng-container *ngFor="let group of displayedValues | listGroupPipe:groupFn | keyvalue; let groupIndex = index">
+                    <li fd-list-group-header>
+                        {{group.key}}
+                    </li>
+                    <li *ngFor="let term of group.value; let termIndex = index;"
+                        (click)="onMenuClickHandler((groupIndex *  group.value.length) + termIndex)"
+                        (keydown)="onListKeydownHandler($event, (groupIndex *  group.value.length) + termIndex)"
+                        fd-list-item
+                        tabindex="0"
+                        class="fd-combobox-list-item"
+                        [selected]="displayFn(term) === inputText">
+                        <ng-container [ngTemplateOutlet]="itemH" [ngTemplateOutletContext]="{term: term}"></ng-container>
+                    </li>
+                </ng-container>
+            </ng-container>
+            <ng-container *ngIf="!groupFn">
                 <li *ngFor="let term of displayedValues; let index = index;"
                     (click)="onMenuClickHandler(index)"
-                    (keydown)="onMenuKeydownHandler($event, index)"
-                    fd-menu-item
-                    tabindex="0">
-                    <span *ngIf="!itemTemplate"
-                          [innerHTML]="term | displayFnPipe:displayFn | highlight:inputText:highlighting"
-                    ></span>
-                    <ng-container *ngIf="itemTemplate">
-                        <ng-container *ngTemplateOutlet="itemTemplate; context: {$implicit: term}"></ng-container>
-                    </ng-container>
+                    (keydown)="onListKeydownHandler($event, index)"
+                    fd-list-item
+                    tabindex="0"
+                    class="fd-combobox-list-item"
+                    [selected]="displayFn(term) === inputText">
+                    <ng-container [ngTemplateOutlet]="itemH" [ngTemplateOutletContext]="{term: term}"></ng-container>
                 </li>
-            </ul>
-        </fd-menu>
+            </ng-container>
+        </ul>
     </fd-popover-body>
 </fd-popover>
+
+<ng-template let-term="term" #itemH>
+    <span *ngIf="!itemTemplate"
```

Comment:
I'd consider using `*ngIf="... ; else"` syntax, but current implementation is also fine.